File: apps/users/schemas/google.py
from ninja import Schema
from typing import List, Optional

class GeminiModelSchema(Schema):
    id: str
    display_name: Optional[str] = ""
    description: Optional[str] = ""
    input_token_limit: Optional[int] = 0
    output_token_limit: Optional[int] = 0

class GoogleStatusSchema(Schema):
    status: str
    message: str
    models: Optional[List[GeminiModelSchema]] = None

class UsageMetricsSchema(Schema):
    status: str
    message: str
    tracked_locally: bool
    request_count: Optional[int] = None

File: frontend/src/api/google.ts
import apiClient from './client';

export interface GeminiModel {
  id: string;
  display_name: string;
  description: string;
  input_token_limit: number;
  output_token_limit: number;
}

export interface GoogleStatus {
  status: 'success' | 'error';
  message: string;
  models?: GeminiModel[];
}

export interface UsageMetrics {
  status: string;
  message: string;
  tracked_locally: boolean;
  request_count?: number;
}

export const googleApi = {
  getStatus: async (): Promise<GoogleStatus> => {
    return apiClient.get('/google/status');
  },
  
  getMetrics: async (): Promise<UsageMetrics> => {
    return apiClient.get('/google/metrics');
  }
};
